src/sentry/options/manager.py

```diff
@@ -45,6 +45,38 @@ def choices(cls) -> Sequence[Tuple[str, str]]:
         return [(i.name, i.value) for i in cls]
 
 
+class NotWritableReason(Enum):
+    """
+    Represent the reason that prevents us from attempting an update
+    of an option on a specific UpdateChannel.
+    """
+
+    # The option is registered with the FLAG_PRIORITIZE_DISK flag and it is
+    # also stored on disk as part of sentry settings. Nobody can update this.
+    OPTION_ON_DISK = "option_on_disk"
```

Comment:
I need clarity around this.  I see some options configured with "prioritize disk" and "remote modifiable" e.g. the slack option: `register("slack.client-id", flags=FLAG_PRIORITIZE_DISK | FLAG_AUTOMATOR_MODIFIABLE)`.  Is this an impossible state?  I'm still reading through the PR so this detail may reveal itself to me.

Edit: maybe the "and" in the comment is the critical condition?  If "prioritize disk" and the option is stored on disk in `[...]/server/conf.py`?

Reply:
It is not an impossible state (regrettably). 
`PRIORITIZE_DISK` means: trust the settings and prevent it from being stored in the DB IF the option is actually set in the settings file. 
If it is not set in the settings file, then the option can be remotely changed. 
This is by design I believe.